### PR TITLE
Reader: Add support for the Post Carousel and Jetpack Slideshow blocks

### PR DESCRIPTION
**Pull request by sixhours**
Related to #81255 and #45085

## Proposed Changes

* Treat the Post Carousel and Jetpack Slideshows like embeds in the Reader (and other embed contexts, like inline help)
* When we find an instance of a post carousel block or a Jetpack Slideshow block in a post, wrap the markup in a `DotPager` component.
* I experimented with using Swiper (which is the library originally powering these blocks) but opted for DotPager for simplicity and consistency with other carousels (like gallery posts) in the Reader.
* Adds some carousel-specific styles.

**Before**

Screenshot 2023-12-04 at 2 31 47 PM

**After**

Screenshot 2023-12-04 at 2 30 55 PM

## Testing Instructions

* Switch to this PR
* Navigate to `/read`
* Find a post (I had to publish one) with a Post Carousel block containing several posts.
* View the full post; you should see a working carousel rather than an unformatted list of post details.
* Find or publish a post with a Jetpack Slideshow and view it in the Reader (full post view)
* You should see a slideshow instead of a list of images in the full post view.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/blocks/reader-full-post/carousel.scss
Comment (daledupreez) [inline client/blocks/reader-full-post/carousel.scss]: Nit: can we please use the full extension here so simple _Find_ operations on the selector will work?
```suggestion
	.wp-block-newspack-blocks-carousel__placeholder {
```

File: client/blocks/reader-full-post/carousel.scss
Comment (daledupreez) [inline client/blocks/reader-full-post/carousel.scss]: There is going to be some overlap/bleed between these selectors and those under `div .entry-title a` on lines 23-34. Is that intentional? Is there a way for us to be more specific? If it's intentional, and we don't want to be more specific, we should add some explicit comments to that effect.

EDIT: I now see more selectors below that will also apply to the items that also have the `div` intermediate selector. This feels like it's messy and may be hard to follow.

File: client/components/embed-container/index.jsx
Comment (daledupreez) [inline client/components/embed-container/index.jsx]: Nit: should we also require `carouselItems` to have a length?
```suggestion
		if ( carouselItems && carouselItems.length ) {
```

File: client/components/embed-container/index.jsx
Comment (daledupreez) [inline client/components/embed-container/index.jsx]: Nit: `index` is likely to be only an integer, so we'll end up with duplicate `id` attributes in cases where we have multiple carousels on the same page. Unless we have some handle on the blog ID and/or post ID _and_ the positional details of the carousel within the post (i.e. is this the _n_th carousel?), it feels risky to use an `id` for this element.

File: client/components/embed-container/index.jsx
Comment (daledupreez) [inline client/components/embed-container/index.jsx]: This use of an object and then a spread operator for all the fields feels unexpected and unusual to me. Could you comment on why you're doing it this way? (I feel like I am missing something!)

I would normally expect something more like the following:
```suggestion
						return (
							<div
								key={ index }
								className={ classNames( 'carousel-slide', item?.className ) }
								dangerouslySetInnerHTML={ __html: item?.innerHTML }
							/>;
```

**Review by daledupreez — COMMENTED** [review]
From a functional perspective, this is working as described, but I noticed some small DOM-related items that we should probably address before merging.

File: client/blocks/reader-full-post/carousel.scss
Comment (sixhours) [reply client/blocks/reader-full-post/carousel.scss]: Ahh nice catch, I didn't notice the weird nested/duplicate selectors. I've cleaned this up a bit and made the `div` more specific!

File: client/components/embed-container/index.jsx
Comment (sixhours) [reply client/components/embed-container/index.jsx]: Ahh good call, I think I meant to use `key` instead of `id` to suppress the child-items list warning. We don't need the ID AFAIK.

File: client/components/embed-container/index.jsx
Comment (sixhours) [reply client/components/embed-container/index.jsx]: 🤦‍♀️ At one point I had refactored this to use `createElement` so we could change the main class from a DIV to whatever the original item class was, but changed my mind and forgot to switch it back to the standard format here. Fixed. :)

**Review by daledupreez — APPROVED** [review]
Thanks for cleaning that up! I am sure some future maintainer will be _much_ happier with the CSS all in one tree!

I've retested, including at mobile screen widths, and things look good! 🚢

**Pull request merged**
